portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -0,0 +1,91 @@
+package com.google.sps.servlets;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.lang.System;
+
+@WebServlet("/user/*")
+public final class AuthenticationServlet extends HttpServlet {
+    private UserService userService = UserServiceFactory.getUserService();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String uri = request.getRequestURI(); // see below - we'll be removed after the review
+
+        switch(this.getActionPath(request.getRequestURI())) { // REVIEW: is this acceptable? or shall I use uri above?
```

Comment:
I think this is fine. Lets see what Maksim thinks

Reply:
Not critical to my, but splitting into two is easier to read.

Reply:
But switch statement can be a map "action" -> Action handler ;-)

Just an observation not a request.

Reply:
@semenovm you mean the Map data structure? You query it by .get(url) and then you get the appropriate handler?